### PR TITLE
To-do-list Interactive list

### PR DESCRIPTION
**Pull request by MariamOsman11**
# What I did

- Add a new JavaScript file and import it as a module:
- it will contain methods related to the status updates .
- Add event listener to the checkbox.
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -59,4 +59,29 @@ listContainer.addEventListener('click', (e) => {
...
+      const Info = JSON.parse(localStorage.getItem('TasksInfo'));
+      Info[checkStatus.id - 1].completed = true;
+      localStorage.setItem('TasksInfo', JSON.stringify(Info));
+      checkStatus.classList.add('Complete');
+    } else {
+      const Info = JSON.parse(localStorage.getItem('TasksInfo'));
+      Info[checkStatus.id - 1].completed = false;
+      localStorage.setItem('TasksInfo', JSON.stringify(Info));
+      checkStatus.classList.remove('Complete');
+    }
```
Comment (laiifuu) [inline src/index.js]: Functionality related to the tasks' status update should be in its own file and in order to use it you have to import the file as a module as stated in the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Add%20a%20new,/%20false).

I also suggest you add an event listener of type 'change' to the checkbox to update the status of the tasks.

**Review by laiifuu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :stop_sign: 

## Hi @MariamOsman11  :wave: 

Good job so far on your To-do list project!
There are some issues that you still need to work on to go to the next project but you are almost there!
so-close

## Highlights
- [x] Descriptive pull request :heavy_check_mark: 

## Required Changes ♻️
- [ ] When removing a task, the local storage is being updated as it should as well as the id of each task, but the ids of the tasks in the html are not, which causes a problem when deleting tasks consecutively: 
     - I started by adding 4 tasks: 
         - The html: 
          Screenshot from 2023-06-17 16-03-55
         - The local storage:
         Screenshot from 2023-06-17 16-05-01
    - I deleted task 1 which resulted in the following: 
            - The html: 
            Screenshot from 2023-06-17 16-06-01
            - The local storage: 
            Screenshot from 2023-06-17 16-06-14
    - I deleted then task 2 which resulted in this: 
             - The html: 
             Screenshot from 2023-06-17 16-07-03
             - The local storage: 
             Screenshot from 2023-06-17 16-07-19

The end result in the local storage should have been task 3 and 4, not 2 and 4. 
Kindly fix this issue.

- [ ] Nice work on the UI :+1: However, there are some slight issues you can fix: 
Screenshot from 2023-06-17 17-18-31
The checkbox, the task text and the icons should be perfectly alligned, as well as the input text and the plus icon


_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (MariamOsman11) [reply src/index.js]: @laiifuu 
I started inserting the numbers and select first one

 then deleted the first one. my local storage is showing ids 123

 then the ids in the element is showing 123


I don't think I have an issue when viewing from my computer.kindly reconsider your review.Thanks

Comment (MariamOsman11) [reply src/index.js]: @laiifuu  kindly delete the data and refresh to see it, if it's okay for you.

Comment (laiifuu) [reply src/index.js]: Hey @MariamOsman11 :wave: Do you mean everytime a task is deleted, I should refresh the page ?

Comment (MariamOsman11) [reply src/index.js]: @laiifuu nope, I mean mine is working as intended.

Comment (laiifuu) [reply src/index.js]: @MariamOsman11 do you think you can join me in a quick zoom call ?

Comment (MariamOsman11) [reply src/index.js]: @laiifuu yes,call me now

Comment (peterrobert) [reply src/index.js]: This change has been done well and everything is working well

**Review by peterrobert — APPROVED** [review]
Hi @MariamOsman11, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work on your design it looks awesome ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by laiifuu** [comment]
Congrats @MariamOsman11 🥳🥳

**Comment by MariamOsman11** [comment]
@laiifuu  Thank you so much Hayet!
I really appreciated your help!

**Comment by laiifuu** [comment]
No problem 😄 I wish you the best of luck in the future ✌️

**Pull request merged**